presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/ReduceAggStateSerializer.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.operator.aggregation.ReduceAggregationFunction;
+import com.facebook.presto.spi.function.AccumulatorStateSerializer;
+
+public class ReduceAggStateSerializer
+        implements AccumulatorStateSerializer<ReduceAggState>
+{
+    private final Type type;
+
+    public ReduceAggStateSerializer(Type type)
+    {
+        this.type = type;
+    }
+
+    @Override
+    public Type getSerializedType()
+    {
+        return type;
+    }
+
+    @Override
+    public void serialize(ReduceAggState state, BlockBuilder blockBuilder)
+    {
+        ReduceAggregationFunction.write(type, state, blockBuilder);
+    }
+
+    @Override
+    public void deserialize(Block block, int index, ReduceAggState state)
+    {
+        if (block.isNull(index)) {
+            state.setValue(null);
+        }
+        else if (type.getJavaType() == long.class) {
+            state.setValue(type.getLong(block, index));
+        }
+        else if (type.getJavaType() == double.class) {
+            state.setValue(type.getDouble(block, index));
+        }
+        else if (type.getJavaType() == boolean.class) {
+            state.setValue(type.getBoolean(block, index));
+        }
+        else if (type.getJavaType() == Block.class) {
+            state.setValue(block.getBlock(index));
+        }
+        else {
+            state.setValue(block.getSlice(index, 0, block.getSliceLength(index)));
+        }
```

Comment:
This could be simplified with `readNativeValue`

Reply:
Done. Somehow I thought we were doing bindTo. Apparently not. I will look into that later.

Reply:
Done. Somehow I thought we were doing bindTo. Apparently not. I will look into that later.